cmd/parser/parse.go

```diff
@@ -0,0 +1,156 @@
+	cobraCmd.Flags().StringVarP(&cmd.ruleFilePath, "file", "f", "", "Read rules from a file pointed to by parameter value. The file must contain a valid yaml list, where each rule entry starts with '-' and is placed in its own line.")
+	cobraCmd.Flags().BoolVarP(&cmd.sort, "priority", "p", false, "Sort rule entries by their priority, in descending priority order.")
+	cobraCmd.Flags().BoolVarP(&cmd.unique, "unique", "u", false, "Display only non duplicated rules. Error entries are not considered for uniqueness.")
+	cobraCmd.Flags().BoolVarP(&cmd.signature, "signature", "s", false, "Mark rules with the mirrored signatures as duplicated. For example aws(PR=*, HR=westeurope) and aws(PR=westeurope, HR=*) are considered duplicated because of having mirrored signatures.")
+	cobraCmd.Flags().BoolVarP(&cmd.noColor, "no-color", "n", false, "Disable use color characters when generating output.")
+	cobraCmd.MarkFlagsOneRequired("entry", "file")
+
+	return cobraCmd
+}
+
+type ProcessingPair struct {
+	ParsingResults  *rules.ParsingResult
+	MatchingResults *rules.MatchingResult
+}
+
+func (cmd *ParseCommand) Run() error {
+
+	printer := rules.NewColored(cmd.cobraCmd.Printf)
+	if cmd.noColor {
+		printer = rules.NewNoColor(cmd.cobraCmd.Printf)
+	}
+
+	if cmd.match != "" && (!cmd.sort || !cmd.unique) {
+		cmd.cobraCmd.Printf("\tMatching is only supported when both priority and uniqueness flags are specified.\n")
+		return nil
+	}
+
+	// create enabled plans
+	enabledPlans := broker.EnablePlans{}
+	for _, plan := range broker.PlanNamesMapping { 
+		enabledPlans = append(enabledPlans, plan)
+	}
+
+	var rulesService *rules.RulesService
+	var err error
+	if cmd.ruleFilePath != "" {
+		cmd.cobraCmd.Printf("Parsing rules from file: %s\n", cmd.ruleFilePath)
+		rulesService, err = rules.NewRulesServiceFromFile(cmd.ruleFilePath, &enabledPlans, cmd.sort, cmd.unique, cmd.signature)
+	} else {
+		rulesService, err = rules.NewRulesServiceFromString(cmd.rule, &enabledPlans, cmd.sort, cmd.unique, cmd.signature)
+	}
+
+	if err != nil {
+		cmd.cobraCmd.Printf("Error: %s\n", err)
+		return nil
+	}
+
+	var dataForMatching *rules.MatchableAttributes
+	if cmd.match != "" {
+		dataForMatching = getDataForMatching(cmd.match)
+	}
+
+	var matchingResults map[uuid.UUID]*rules.MatchingResult
+	if cmd.match != "" && dataForMatching != nil {
+		matchingResults = rulesService.Match(dataForMatching)
+	}
+
+	printer.Print(rulesService.Parsed.Results, matchingResults)
+
+	hasErrors := false
+	for _, result := range rulesService.Parsed.Results {
+		if result.HasErrors() {
+			hasErrors = true
+			break
+		}
+	}
+
+	if hasErrors {
+		cmd.cobraCmd.Printf("There are errors in your rule configuration. Fix above errors in your rule configuration and try again.\n")
+		return nil
+	}
+
+	return nil
+}
+
+type conf struct {
+	Rules []string `yaml:"rule"`
+}
+
+func getDataForMatching(content string) *rules.MatchableAttributes {
+	testData := &rules.MatchableAttributes{}
```

Comment:
`testData` seem out of place here.

Reply:
https://github.com/ralikio/kyma-environment-broker/commit/a01d377311038469263b12a58c26ef9533e03655